### PR TITLE
Chairs/Beds Repathing/Refactor

### PR DESCRIPTION
**Pull request by AnturK**
- Splits beds and chairs into /obj/structure/bed and obj/structure/chair. I believe the shared functionality is minimal enough so the resulting code duplication is not horrible and will make it easier to differentiate these in the future.
- Adds generic deconstruct helper to obj/structure.
- Moves chair sprites from objects.dmi to chairs.dmi
- Makes stools subtype of chair
- Removes some redundant checks and unused code.
- Fixes passing proper value attack_type for hit reaction for unarmed attacks.

Some new stuff:
- Chairs/Stools can be picked up by dragging them on your sprite.
- 8 damage, they block unarmed attacks with 50% chance.

This is path change hell. I'm open to discussion if the actual path change is needed since i'm not entirely sure myself.

### CODE REVIEW

**Comment by Ultimate-Chimera** [comment]
:+1:

File: code/game/objects/structures/beds_chairs/chair.dm [hunk 1]
```diff
@@ -109,41 +142,99 @@
...
+	desc = "bar brawl essential"
+	w_class = 5
+	force = 8
+	throwforce = 10
+	throw_range = 3
+
+	var/origin_type = /obj/structure/chair
+
+/obj/item/chair/New(loc,obj/structure/chair/origin)
+	name = origin.name
```
Comment (RemieRichards) [inline code/game/objects/structures/beds_chairs/chair.dm]: where's ```..()``` you poo

Comment (AnturK) [inline code/game/objects/structures/beds_chairs/chair.dm]: Whoops.

**Comment by Zilenan91** [comment]
Can you buff their throwforce to around 15 because throwing chairs at people as a viable tactic sounds hilarious.

**Comment by KazeEspada** [comment]
Toolboxes vs Chairs....

Meanwhile the truly robust throws his stack of tiles.

**Comment by duncathan** [comment]
If you're gonna do this please actually update the maps

**Comment by duncathan** [comment]
Oh you already did :+1:

File: code/modules/power/cable.dm [hunk 2]
```diff
@@ -490,7 +490,7 @@ var/global/list/datum/stack_recipe/cable_coil_recipes = list ( \
 	update_icon()
 
 /obj/item/stack/cable_coil/suicide_act(mob/user)
-	if(locate(/obj/structure/bed/stool) in user.loc)
+	if(locate(/obj/structure/chair/stool) in user.loc)
```
Comment (neersighted) [inline code/modules/power/cable.dm]: get_turf?

File: code/game/objects/structures/electricchair.dm [hunk 3]
```diff
@@ -1,18 +1,18 @@
...
 	return
 
-/obj/structure/bed/chair/e_chair/attackby(obj/item/weapon/W, mob/user, params)
+/obj/structure/chair/e_chair/attackby(obj/item/weapon/W, mob/user, params)
 	if(istype(W, /obj/item/weapon/wrench))
-		var/obj/structure/bed/chair/C = new /obj/structure/bed/chair(loc)
+		var/obj/structure/chair/C = new /obj/structure/chair(loc)
 		playsound(loc, 'sound/items/Ratchet.ogg', 50, 1)
 		C.dir = dir
 		part.loc = src.loc
```
Comment (neersighted) [inline code/game/objects/structures/electricchair.dm]: dtv

File: code/game/objects/structures/electricchair.dm [hunk 4]
```diff
@@ -22,13 +22,13 @@
 		return
 	return
 
-/obj/structure/bed/chair/e_chair/rotate()
+/obj/structure/chair/e_chair/rotate()
```
Comment (neersighted) [inline code/game/objects/structures/electricchair.dm]: do overlays not follow their parent's dir?

File: code/game/objects/structures/beds_chairs/chair.dm [hunk 5]
```diff
@@ -109,41 +142,99 @@
...
+		if(!hold_icon || !ishuman(usr) || buckled_mob)
+			return
+		usr.visible_message("<span class='notice'>[usr] grabs \the [src.name].</span>", "<span class='notice'>You grab \the [src.name].</span>")
+		var/obj/item/chair/C = new /obj/item/chair(get_turf(src),src)
+		usr.put_in_hands(C)
+		qdel(src)
+
+/obj/item/chair
+	name = "chair"
+	desc = "bar brawl essential"
```
Comment (neersighted) [inline code/game/objects/structures/beds_chairs/chair.dm]: caps/punctuation?

File: code/game/objects/structures/beds_chairs/chair.dm [hunk 6]
```diff
@@ -47,19 +79,19 @@
...
+/obj/structure/chair/proc/handle_layer()
 	if(dir == NORTH)
-		src.layer = FLY_LAYER
+		layer = FLY_LAYER
 	else
-		src.layer = OBJ_LAYER
+		layer = OBJ_LAYER
 
-/obj/structure/bed/chair/proc/spin()
+/obj/structure/chair/proc/spin()
```
Comment (neersighted) [inline code/game/objects/structures/beds_chairs/chair.dm]: Action button for the buckled mob?

File: code/game/objects/structures/beds_chairs/chair.dm [hunk 7]
```diff
@@ -47,19 +79,19 @@
...
 	src.dir = turn(src.dir, 90)
 	handle_layer()
 	if(buckled_mob)
 		buckled_mob.dir = dir
 
-/obj/structure/bed/chair/verb/rotate()
+/obj/structure/chair/verb/rotate()
```
Comment (neersighted) [inline code/game/objects/structures/beds_chairs/chair.dm]: Just replace this with alt-click?

**Comment by PKPenguin321** [comment]
:+1:

**Comment by WJohn** [comment]
You sure do like fighting map conflicts, don't you?

**Comment by optimumtact** [comment]
It's finally happening!

also @WJohn map conflicts on paths aren't so bad, I usually write a regex to search replace paths, then just checkout the latest maps and rerun the regex and check in the result.

it's only when you have to change layout of stuff that conflicts are a problem.

**Comment by KorPhaeron** [comment]
They need an attack sound.

**Comment by KorPhaeron** [comment]
Oh godamnit I caused conflicts by merging that space battle thing you tricked me @tkdrg

**Comment by tkdrg** [comment]
did i

**Comment by KorPhaeron** [comment]
@AnturK 

Sorry

**Pull request merged**